theano/gof/opt.py

```diff
@@ -1131,13 +1131,22 @@ class LocalMetaOptimizer(LocalOptimizer):
 
     """
 
-    def __init__(self, tracks=None, optimizers=()):
-        self._tracks = tracks
+    def __init__(self, optimizers=()):
+        self._tracks = [x for o in optimizers for x in o.tracks()]
         self.optimizers = list(optimizers)
         self.verbose = config.metaopt.verbose
+        self.track_dict = defaultdict(lambda: [])
```

Comment:
Yes, they are in the following lines. But initializing everything to be empty and then calling `self.register(optimizers)` would reduce code duplication / possible errors.